src/cp/constraints/planed_shifts.py

```diff
@@ -9,67 +9,73 @@
 class PlannedShiftsConstraint(Constraint):
     KEY = "planned-shifts"
 
-    SHIFT_MAPPING = {
-        "F": 0,
-        "Z": 1,
-        "S": 2,
-        "N": 3,
-        "Z60": 4,
-    }
-
-    EXCLUSIVE_SHIFTS = ["Z60"]  # Shifts that can only be assigned with planned_shifts
-
     def __init__(self, employees: list[Employee], days: list, shifts: list[Shift]):
         super().__init__(employees, days, shifts)
 
     def create(self, model: CpModel, variables: dict[str, Variable]):
-        # Collect all employees who are allowed to have Z60 shifts
-        employees_with_z60 = set()
+        # Collect employees with exclusive shifts
+        employees_with_exclusive_shifts = {}
 
+        # Process all planned shifts
         for employee in self._employees:
-            if not hasattr(employee, "_planned_shifts") or not employee._planned_shifts:
-                continue
-
             for day_num, shift_code in employee._planned_shifts:
-                if shift_code in self.EXCLUSIVE_SHIFTS:
-                    employees_with_z60.add(employee.get_key())
+                # Track exclusive shifts
+                if shift_code in Shift.EXCLUSIVE_SHIFTS:
+                    if shift_code not in employees_with_exclusive_shifts:
+                        employees_with_exclusive_shifts[shift_code] = set()
+                    employees_with_exclusive_shifts[shift_code].add(employee.get_key())
 
-                # Existing code für planned shifts...
+                # Find the day
                 day = next((d for d in self._days if d.day == day_num), None)
                 if not day:
+                    logging.warning(f"Day {day_num} not found for planned shift")
                     continue
 
-                shift_id = self.SHIFT_MAPPING.get(shift_code)
+                # Map the shift code to ID
+                shift_id = Shift.SHIFT_MAPPING.get(shift_code)
                 if shift_id is None:
-                    logging.warning(f"Unknown shift code: {shift_code}")
+                    logging.warning(
+                        f"Unknown shift code: {shift_code} for {employee.name}"
+                    )
                     continue
 
+                # Find the shift
                 shift = next((s for s in self._shifts if s.get_id() == shift_id), None)
```

Comment:
This is unreadable, iterate over `self._days` and use timedelta to get other days